chef_master/source/chef_license_accept.rst

```diff
@@ -0,0 +1,277 @@
+
+#. ``CHEF_LICENSE=<value>`` as an environment variable.
+
+``<value>`` can be specified as one of the following:
+
+#. ``accept`` - Accepts the license and attempts to persist a marker file locally. Persisting these marker files means future invocations do not require accepting the license again.
+
+#. ``accept-silent`` - Similar to ``accept`` except no messaging is sent to STDOUT
+
+#. ``accept-no-persist`` - Similar to ``accept-silent`` except no marker file is persisted. Future invocation will require accepting the license again.
+
+If no command line argument or environment variable is set, these products will attempt to request acceptance through an interactive prompt.
+If the prompt cannot be displayed, then the product will fail with an exit code 172.
+
+If the product attempts to persist the accepted license and fails, a message will be sent to STDOUT, but product invocation will continue successfully. 
+In a future invocation, however, the license would need to be accepted again.
+
+Products with specific features or differences from this general behavior are documented below.
+
+Chef Workstation
+-----------------------------------------------------
+Chef Workstation contains multiple chef products.
+When invoking the ``chef`` command line tool and accepting the license, users are required to accept the license for all the embedded products as well.
+The same license applies to all products, but each product must have its own license acceptance.
+``chef --chef-license accept`` will accept the license for Chef Workstation, Chef Infra, Chef InSpec, and Push Jobs Client.
+
+Chef Infra
+-----------------------------------------------------
+In addition to the above methods, users can specify ``chef_license = 'accept'`` in their Chef config.
+On a workstation, this can be specified in ``~/.chef/config.rb`` or ``~/.chef/knife.rb``, and on a node, it can be specified in ``/etc/chef/client.rb``.
+This method of license acceptance is backwards-compatible to non-EULA versions of Chef Infra.
+
+Habitat
+-----------------------------------------------------
+The Habitat license can be accepted by setting either environment variables ``HAB_LICENSE=accept`` or ``HAB_LICENSE=accept-no-persist``, or by executing ``hab license accept``.
+If neither of these are set, the first ``hab`` invocation will ask the user to accept the license through an interactive prompt.
+
+Note: This does not apply to packages installed using Habitat.
+In those cases, a Habitat configuration is set per package.
+Documentation for this is coming soon.
+
+Server Products
+=====================================================
+* Chef Server, which also contains
+    * Push Jobs Server
+* Chef Automate
+* Supermarket
+
+Server products are typically installed and managed by some kind of supervisor instead of a user on a workstation.
+Because of this, there is no interactive option for accepting the license.
```

Comment:
What is "this" referring to in this sentence?

Reply:
Changed that paragraph to

```
Server products are typically installed and managed by some kind of process supervisor. There are no interactive license
acceptance options because process supervisors do not easily allow interactivity. Instead the license is accepted during
the ``reconfigure`` command or ``upgrade`` command for the Omnibus ctl command.
```